src/overview.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Dart overview
+description: A short introduction to Dart
+---
+
+<img style="padding: 30px; float: right; width: 300px" src="{% asset
+logo_lockup_dart_horizontal.png @path %}" alt="Dart product logo">
+
+Dart is a client-optimized language for developing fast apps on any platform.
+Its goal is to offer the most productive programming language for
+multi-platform development, paired with a
+[flexible execution runtime platform](/platforms) for app frameworks.
+
+Languages are defined by their _technical envelope_,
+the many choices that are made during development that
+shape the capabilities and strengths of a language.
+Dart is designed for a technical envelope that is
+particularly suited to client development,
+prioritizing both development (sub-second stateful hot reload) and
+high-quality production experiences across
+a wide variety of compilation targets (web, mobile, and desktop).
+
+Dart also forms the foundation of [Flutter](https://flutter.dev).
+Dart provides the language and runtimes that power Flutter apps,
+but Dart also supports many core developer tasks like
+formatting, analyzing, and testing code.
+
+
+## Dart: The language
+
+The Dart language is type safe;
+it uses static type checking to ensure that
+a variable’s value _always_ matches the variable’s static type.
+Sometimes, this is referred to as sound typing.
+Although types are mandatory,
+type annotations are optional because of type inference.
+The Dart typing system is also flexible,
+allowing the use of a 'dynamic' type combined with runtime checks,
+which can be useful during experimentation or
+for code that needs to be especially dynamic.
+
+Dart offers [sound null safety](/null-safety),
+meaning that values can’t be null unless you say they can be.
+With sound null safety, Dart can protect you from
+null exceptions at runtime through static code analysis.
+Unlike many other null-safe languages,
+when Dart determines that a variable is non-nullable,
+that variable is always non-nullable.
+If you inspect your running code in the debugger,
+you’ll see that non-nullability is retained at runtime
+(hence _sound_ null safety).
+
+This code sample showcases a number of Dart language features:
+libraries, async calls, nullable and non-nullable types,
+arrow syntax, generators, streams, and getters.
+A more comprehensive set of annotated language examples can be found in
+[Dart by Example](http://jpryan.me/dartbyexample/) by
+[@jryanio](https://twitter.com/jryanio).
```

Comment:
Should we be linking to dart.dev/samples instead? We know that'll stay updated.

/cc @johnpryan

Reply:
+1 - I haven't updated Dart By Example. (it's somewhere on my list, but not very high up right now)

Reply:
I just love it though, John -- it's such a comprehensive list of examples (compared to the language tour). FWIW, I'd love to see an extended version of this on the website.

Reply:
John is going to take a stab at combing his content with the current Dart samples content, and maybe even in the process wire the samples up with embedded DartPad 🚀